src/hooks/usePollQuery.ts

```diff
@@ -0,0 +1,140 @@
+  refetch: (
+    variables?: ApolloQueryVariables
+  ) => Promise<ApolloQueryResult<ApolloQueryResultType>>;
+  search: string;
+}
+
+export const usePollQuery = <ApolloQueryVariables, ApolloQueryResultType>({
+  networkStatus,
+  getQueryVariables,
+  refetch,
+  search,
+}: Params<ApolloQueryVariables, ApolloQueryResultType>): {
+  showSkeleton: boolean;
+} => {
+  const { pathname } = useLocation();
+  const [initialPathname] = useState(pathname);
+  const { id: resourceId } = useParams<{ id: string }>();
+  const [intervalId, setIntervalId] = useState<number>();
+  // this variable is true when query variables have changed and the query is loading
+  // this means the user interacted with the table/list filters, sort, or page
+  const [queryVarDiffOccured, setQueryVarDiffOccured] = useState(false);
+  const currentQueryVariables = getQueryVariables(search, resourceId);
+  const prevQueryVariables = usePrevious(currentQueryVariables);
+  const isLoading = isNetworkRequestInFlight(networkStatus);
+
+  // detects when query variables have changed
+  useEffect(() => {
+    if (!isEqual(currentQueryVariables, prevQueryVariables)) {
+      setQueryVarDiffOccured(true);
+    }
+  }, [currentQueryVariables, prevQueryVariables, setQueryVarDiffOccured]);
+
+  // when loading goes from true to false, do not show the loading skeleton
+  useEffect(() => {
+    if (!isLoading) {
+      setQueryVarDiffOccured(false);
+    }
+  }, [isLoading, setQueryVarDiffOccured]);
+
+  // this is to initiate polling for the first time after useQuery
+  // submits it's first request and is not dependent on URL changes
+  useEffect(() => {
+    if (!intervalId) {
+      pollQuery({
+        setIntervalId,
+        refetch,
+        search,
+        getQueryVariables,
+        resourceId,
+      });
+    }
+    return () => clearInterval(intervalId);
+  }, [
+    intervalId,
+    refetch,
+    search,
+    getQueryVariables,
+    resourceId,
+    setIntervalId,
+  ]);
+
+  // clears the interval when the page changes/component unmounts
+  useEffect(() => {
+    if (pathname !== initialPathname) {
+      clearInterval(intervalId);
+    }
+  }, [pathname, initialPathname, intervalId]);
+
+  // reponsible for clearing the current polling query and
+  // starting a new polling query based on new query variables
+  useEffect(() => {
+    if (
+      intervalId &&
+      !isEqual(currentQueryVariables, prevQueryVariables) &&
+      pathname === initialPathname
+    ) {
+      clearInterval(intervalId);
+      refetch(currentQueryVariables);
+      pollQuery({
+        search,
+        resourceId,
+        refetch,
+        setIntervalId,
+        getQueryVariables,
+      });
+    }
+  }, [
+    currentQueryVariables,
+    prevQueryVariables,
+    networkStatus,
+    refetch,
+    intervalId,
+    resourceId,
+    getQueryVariables,
+    search,
+    pathname,
+    initialPathname,
+  ]);
+
+  return {
+    showSkeleton: queryVarDiffOccured,
```

Comment:
If this is the only thing this hook returns i think you can just return `queryVarDiffOccured` directly